app/services/context_service.rb

```diff
@@ -0,0 +1,31 @@
+class ContextService
```

Comment:
@bzwei Should this class itself be called AdminContext.

Reply:
@bzwei What would happen if the RBAC team gives us a well defined role that allows us to access the principals in a tenant, how much of the above code would have to change. We need to note that what we are doing is a hack and not an ideal solution for us. So if we have to throw this away and use a real solution we dont have to redo the whole thing all over.

Reply:
I moved `Request#switch_context` to this class. This class will provide all context switching/restoration operations, not limit to admin. We can continue to use this service class to handle changes when RBAC is improved. Anyway we always need to switch a role/context to access groups/principals because they definitely will not be accessible to a regular user.